docset/winserver2022-ps/netsecurity/New-NetFirewallDynamicKeywordAddress.md

```diff
@@ -0,0 +1,212 @@
+---
+description: The New-NetFirewallDynamicKeywordAddress cmdlet creates a dynamic keyword address.
+external help file: NetFirewallDynamicKeywordAddress.cmdletDefinition.cdxml-help.xml
+Module Name: NetSecurity
+ms.date: 9/20/2021
+online version: https://docs.microsoft.com/powershell/module/netsecurity/new-netfirewalldynamickeywordaddress?view=windowsserver2022-ps&wt.mc_id=ps-gethelp
+schema: 2.0.0
+title: New-NetFirewallDynamicKeywordAddress
+---
+
+# New-NetFirewallDynamicKeywordAddress
+
+## SYNOPSIS
+Creates a dynamic keyword address.
+
+## SYNTAX
+
+```
+New-NetFirewallDynamicKeywordAddress [-Id <String>] [-Keyword <String>] [-Addresses <String>]
+ [-AutoResolve <Boolean>] [-CimSession <CimSession[]>] [-ThrottleLimit <Int32>] [-AsJob] [-WhatIf] [-Confirm]
+ [<CommonParameters>]
+```
+
+## DESCRIPTION
+The **New-NetFirewallDynamicKeywordAddress** cmdlet creates a dynamic keyword address.
+
+A firewall rule can use dynamic keyword addresses instead of explicitly defining IP addresses for its remote address condition.
+
+## EXAMPLES
+
+### Example 1: Create dynamic keyword address with AutoResolve
+```powershell
+ New-NetFirewallDynamicKeywordAddress -Id 258741ed-94a3-406b-9b0c-81fb145a4592 -Keyword ContosoSubnet -AutoResolve
+```
+
+This command creates an AutoResolve dynamic keyword address.
+
+Be sure to generate your own GUID.
```

Comment:
If I leave the -Id parameter out, will a GUID be generated on my behalf? It seems odd to require a GUID be generated and then provided. If we truly require that, perhaps we should reference a GUID-generating cmdlet to use in-line here and elsewhere where we instruct the reader to bring their own GUID.

Reply:
@mstojens I got these two examples from here: https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-fasp/83c04df4-314b-444e-972a-85ff3c6e9726

The -id param is optional. Unless there's another way to identify a dynamic keyword address, it must get an ID somewhere. 

I don't want to say that I was guessing down in the param descriptions, but can you verify that it actually works that way?

Reply:
Ok so I verified that the following will work:

> New-NetFirewallDynamicKeywordAddress -Id "{258741ED-94A3-406B-9B0C-81FB145A4592}" -Keyword "TestCmdlet" -AutoResolve $True

The GUID needs to be wrapped in quotes and curly brackets, Keyword need to be a string, and AutoResolve needs to use PowerShell boolean formats ($True, $False, 0, 1)